server/channels/store/searchlayer/layer.go

```diff
@@ -111,6 +111,32 @@ func (s *SearchStore) indexUser(rctx request.CTX, user *model.User) {
 	}
 }
 
+func (s *SearchStore) indexChannelsForTeam(rctx request.CTX, teamID string) {
+	var (
+		page     int
+		perPage  = 100
+		channels []*model.Channel
+	)
+
+	for {
```

Comment:
Unrelated to this PR but seems unnecessary to index all public channels of a team when a user joins or leaves a team. Instead we could index only the channel they joined by default (town Square, Off Topic and other channels defined in the config), and the same for only the channels a user was member of when they left.

Just curious as to the reasoning for this.